# **Task 2 -  Extract LinkedIn Profiles and Identify Misconduct**


In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv("GMT20240125-121850_RecordingnewChat - Cop.txt", sep="\t", header=None, engine='python')

In [3]:
df

0                         1  \
0     00:15:43             Harsh Sharma:   
1     00:15:45    BADISHAGANDU RAVINDER:   
2     00:15:48               Kavin Puri:   
3     00:15:48             ANIL DEVALLA:   
4     00:15:49              syed sameer:   
...        ...                       ...   
3444  01:41:52  Gurram Ruthvi Chaitanya:   
3445  01:41:53          Rahul Janjirala:   
3446  01:41:53                Vyom Modi:   
3447  01:41:53     Mohammed Abdul Rafay:   
3448  01:41:54       Gayathri Kondepudi:   

                                                      2  
0                                          Discord link  
1                                           very exited  
2     What is the point of making discord group if a...  
3                         https://discord.gg/cSXvEw6Hxd  
4             =-0-=—;’;’;[p’’’;/“?;.’[;/'?;./";.;.";'÷?  
...                                                 ...  
3444                      sir it is showing preequation  
3445                             innomatics hd logo sir  
3446  sir should we add experience related to other ...  
3447                                   sir discord link  
3448                   lms access mail not received sir  

[3449 rows x 3 columns]

In [4]:
df1= pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy(1).txt", sep="\t", header=None, engine='python')

In [5]:
df1

0                           1  \
0     00:15:43               Harsh Sharma:   
1     00:19:47                Adepu Kavya:   
2     00:19:47  Sai Ramya Manthri Pragada:   
3     00:19:47             Jai Chaudhrani:   
4     00:19:47                 Adnan Baig:   
...        ...                         ...   
3659  01:41:52    Gurram Ruthvi Chaitanya:   
3660  01:41:53            Rahul Janjirala:   
3661  01:41:53                  Vyom Modi:   
3662  01:41:53       Mohammed Abdul Rafay:   
3663  01:41:54         Gayathri Kondepudi:   

                                                      2  
0                                          Discord link  
1                                                   yes  
2                                                   yes  
3                                               Audible  
4                                                   Yes  
...                                                 ...  
3659                      sir it is showing preequation  
3660                             innomatics hd logo sir  
3661  sir should we add experience related to other ...  
3662                                   sir discord link  
3663                   lms access mail not received sir  

[3664 rows x 3 columns]

In [6]:
df2=pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (3).txt", sep="\t", header=None, engine='python')

In [7]:
df2

0                         1  \
0     00:15:43             Harsh Sharma:   
1     00:15:45    BADISHAGANDU RAVINDER:   
2     00:15:48               Kavin Puri:   
3     00:15:48             ANIL DEVALLA:   
4     00:15:49              syed sameer:   
...        ...                       ...   
3886  01:41:52  Gurram Ruthvi Chaitanya:   
3887  01:41:53          Rahul Janjirala:   
3888  01:41:53                Vyom Modi:   
3889  01:41:53     Mohammed Abdul Rafay:   
3890  01:41:54       Gayathri Kondepudi:   

                                                      2  
0                                          Discord link  
1                                           very exited  
2     What is the point of making discord group if a...  
3                         https://discord.gg/cSXvEw6Hxd  
4             =-0-=—;’;’;[p’’’;/“?;.’[;/'?;./";.;.";'÷?  
...                                                 ...  
3886                      sir it is showing preequation  
3887                             innomatics hd logo sir  
3888  sir should we add experience related to other ...  
3889                                   sir discord link  
3890                   lms access mail not received sir  

[3891 rows x 3 columns]

In [8]:
df3 = pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (4).txt", sep="\t", header=None, engine='python')

In [9]:
df3

0                       1  \
0     00:15:43           Harsh Sharma:   
1     00:15:45  BADISHAGANDU RAVINDER:   
2     00:15:48             Kavin Puri:   
3     00:15:48           ANIL DEVALLA:   
4     00:15:49            syed sameer:   
...        ...                     ...   
3574  01:37:14     Pavan kalyan Kolla:   
3575  01:37:14          Raju lakkarsu:   
3576  01:37:15    DUDYALA PAVAN KUMAR:   
3577  01:37:16         PRATIK PATIDAR:   
3578  01:37:21    Manaswini Deshpande:   

                                                      2  
0                                          Discord link  
1                                           very exited  
2     What is the point of making discord group if a...  
3                         https://discord.gg/cSXvEw6Hxd  
4             =-0-=—;’;’;[p’’’;/“?;.’[;/'?;./";.;.";'÷?  
...                                                 ...  
3574  Now a days I faced mostly are fake job posts h...  
3575                                                  s  
3576                                                yes  
3577        Reacted to "Now a days I faced m..." with 👍  
3578                                 dat was my dbt too  

[3579 rows x 3 columns]

In [10]:
df4=pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (5).txt", sep="\t", header=None, engine='python')

In [11]:
df4

0                       1  \
0     00:15:43           Harsh Sharma:   
1     00:15:45  BADISHAGANDU RAVINDER:   
2     00:15:48             Kavin Puri:   
3     00:15:48           ANIL DEVALLA:   
4     00:15:49            syed sameer:   
...        ...                     ...   
1508  00:45:16          Krishna Kumar:   
1509  00:45:16          pavansai gude:   
1510  00:45:16         Gudipati Umesh:   
1511  00:45:17    Lavanya Varikuppala:   
1512  00:45:17      suhan reddy paidi:   

                                                      2  
0                                          Discord link  
1                                           very exited  
2     What is the point of making discord group if a...  
3                         https://discord.gg/cSXvEw6Hxd  
4             =-0-=—;’;’;[p’’’;/“?;.’[;/'?;./";.;.";'÷?  
...                                                 ...  
1508                                     Initial letter  
1509                                            capital  
1510                                    Capital letters  
1511                              All are small letters  
1512                               first letter capital  

[1513 rows x 3 columns]

In [12]:
df5  = pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (6).txt", sep="\t", header=None, engine='python')

In [13]:
df5

0                         1  \
0     00:15:43             Harsh Sharma:   
1     00:15:45    BADISHAGANDU RAVINDER:   
2     00:15:48               Kavin Puri:   
3     00:15:48             ANIL DEVALLA:   
4     00:15:49              syed sameer:   
...        ...                       ...   
3563  01:41:52  Gurram Ruthvi Chaitanya:   
3564  01:41:53          Rahul Janjirala:   
3565  01:41:53                Vyom Modi:   
3566  01:41:53     Mohammed Abdul Rafay:   
3567  01:41:54       Gayathri Kondepudi:   

                                                      2  
0                                          Discord link  
1                                           very exited  
2     What is the point of making discord group if a...  
3                         https://discord.gg/cSXvEw6Hxd  
4             =-0-=—;’;’;[p’’’;/“?;.’[;/'?;./";.;.";'÷?  
...                                                 ...  
3563                      sir it is showing preequation  
3564                             innomatics hd logo sir  
3565  sir should we add experience related to other ...  
3566                                   sir discord link  
3567                   lms access mail not received sir  

[3568 rows x 3 columns]

In [14]:
df6 = pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (7).txt", sep="\t", header=None, engine='python')

In [15]:
df6

0                          1  \
0     00:45:20          Ragamayee Aeluri:   
1     00:45:21          Voleti Vaishnavi:   
2     00:45:22         Salikanti Karthik:   
3     00:45:24  laxmi prasanna_IN1240514:   
4     00:45:25           Rizwana Yasmeen:   
...        ...                        ...   
1707  01:41:52   Gurram Ruthvi Chaitanya:   
1708  01:41:53           Rahul Janjirala:   
1709  01:41:53                 Vyom Modi:   
1710  01:41:53      Mohammed Abdul Rafay:   
1711  01:41:54        Gayathri Kondepudi:   

                                                      2  
0                                        case sensitive  
1                                        Capital letter  
2                                     Proper convention  
3                                               capital  
4                                         no Title case  
...                                                 ...  
1707                      sir it is showing preequation  
1708                             innomatics hd logo sir  
1709  sir should we add experience related to other ...  
1710                                   sir discord link  
1711                   lms access mail not received sir  

[1712 rows x 3 columns]

In [16]:
df7 = pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (8).txt", sep="\t", header=None, engine='python')

In [17]:
df7

0                         1  \
0     00:15:43             Harsh Sharma:   
1     00:15:45    BADISHAGANDU RAVINDER:   
2     00:15:48               Kavin Puri:   
3     00:15:48             ANIL DEVALLA:   
4     00:15:49              syed sameer:   
...        ...                       ...   
3874  01:41:52  Gurram Ruthvi Chaitanya:   
3875  01:41:53          Rahul Janjirala:   
3876  01:41:53                Vyom Modi:   
3877  01:41:53     Mohammed Abdul Rafay:   
3878  01:41:54       Gayathri Kondepudi:   

                                                      2  
0                                          Discord link  
1                                           very exited  
2     What is the point of making discord group if a...  
3                         https://discord.gg/cSXvEw6Hxd  
4             =-0-=—;’;’;[p’’’;/“?;.’[;/'?;./";.;.";'÷?  
...                                                 ...  
3874                      sir it is showing preequation  
3875                             innomatics hd logo sir  
3876  sir should we add experience related to other ...  
3877                                   sir discord link  
3878                   lms access mail not received sir  

[3879 rows x 3 columns]

In [18]:
df8 = pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (9).txt", sep="\t", header=None, engine='python')

In [19]:
df8

0                         1  \
0     00:43:23        KARRI DHARMA TEJA:   
1     00:43:23     Shrikant Suryawanshi:   
2     00:43:23         Nishant Dandwate:   
3     00:43:23                Vyom Modi:   
4     00:43:23                Ajay Koka:   
...        ...                       ...   
1996  01:41:52  Gurram Ruthvi Chaitanya:   
1997  01:41:53          Rahul Janjirala:   
1998  01:41:53                Vyom Modi:   
1999  01:41:53     Mohammed Abdul Rafay:   
2000  01:41:54       Gayathri Kondepudi:   

                                                      2  
0     https://www.linkedin.com/in/karri-dharma-teja-...  
1     https://www.linkedin.com/in/shrikant-suryawans...  
2       https://www.linkedin.com/in/nishantdandwate000/  
3                https://www.linkedin.com/in/vyom-modi/  
4      https://www.linkedin.com/in/ajay-koka-94879b21b/  
...                                                 ...  
1996                      sir it is showing preequation  
1997                             innomatics hd logo sir  
1998  sir should we add experience related to other ...  
1999                                   sir discord link  
2000                   lms access mail not received sir  

[2001 rows x 3 columns]

In [20]:
df9 = pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (10).txt", sep="\t", header=None, engine='python')

In [21]:
df9

0                         1  \
0     00:15:43             Harsh Sharma:   
1     00:15:45    BADISHAGANDU RAVINDER:   
2     00:15:48               Kavin Puri:   
3     00:15:48             ANIL DEVALLA:   
4     00:15:49              syed sameer:   
...        ...                       ...   
2603  01:41:52  Gurram Ruthvi Chaitanya:   
2604  01:41:53          Rahul Janjirala:   
2605  01:41:53                Vyom Modi:   
2606  01:41:53     Mohammed Abdul Rafay:   
2607  01:41:54       Gayathri Kondepudi:   

                                                      2  
0                                          Discord link  
1                                           very exited  
2     What is the point of making discord group if a...  
3                         https://discord.gg/cSXvEw6Hxd  
4             =-0-=—;’;’;[p’’’;/“?;.’[;/'?;./";.;.";'÷?  
...                                                 ...  
2603                      sir it is showing preequation  
2604                             innomatics hd logo sir  
2605  sir should we add experience related to other ...  
2606                                   sir discord link  
2607                   lms access mail not received sir  

[2608 rows x 3 columns]

In [22]:
df10 = pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (11).txt", sep="\t", header=None, engine='python')

In [23]:
df10

0                         1  \
0     00:15:43             Harsh Sharma:   
1     00:15:45    BADISHAGANDU RAVINDER:   
2     00:15:48               Kavin Puri:   
3     00:15:48             ANIL DEVALLA:   
4     00:15:49              syed sameer:   
...        ...                       ...   
3859  01:41:52  Gurram Ruthvi Chaitanya:   
3860  01:41:53          Rahul Janjirala:   
3861  01:41:53                Vyom Modi:   
3862  01:41:53     Mohammed Abdul Rafay:   
3863  01:41:54       Gayathri Kondepudi:   

                                                      2  
0                                          Discord link  
1                                           very exited  
2     What is the point of making discord group if a...  
3                         https://discord.gg/cSXvEw6Hxd  
4             =-0-=—;’;’;[p’’’;/“?;.’[;/'?;./";.;.";'÷?  
...                                                 ...  
3859                      sir it is showing preequation  
3860                             innomatics hd logo sir  
3861  sir should we add experience related to other ...  
3862                                   sir discord link  
3863                   lms access mail not received sir  

[3864 rows x 3 columns]

In [24]:
df11 = pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (12).txt", sep="\t", header=None, engine='python')

In [25]:
df11

0                         1  \
0     00:15:43             Harsh Sharma:   
1     00:15:45    BADISHAGANDU RAVINDER:   
2     00:15:48               Kavin Puri:   
3     00:15:48             ANIL DEVALLA:   
4     00:15:49              syed sameer:   
...        ...                       ...   
3629  01:41:52  Gurram Ruthvi Chaitanya:   
3630  01:41:53          Rahul Janjirala:   
3631  01:41:53                Vyom Modi:   
3632  01:41:53     Mohammed Abdul Rafay:   
3633  01:41:54       Gayathri Kondepudi:   

                                                      2  
0                                          Discord link  
1                                           very exited  
2     What is the point of making discord group if a...  
3                         https://discord.gg/cSXvEw6Hxd  
4             =-0-=—;’;’;[p’’’;/“?;.’[;/'?;./";.;.";'÷?  
...                                                 ...  
3629                      sir it is showing preequation  
3630                             innomatics hd logo sir  
3631  sir should we add experience related to other ...  
3632                                   sir discord link  
3633                   lms access mail not received sir  

[3634 rows x 3 columns]

In [26]:
df12 =  pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (13).txt", sep="\t", header=None, engine='python')

In [27]:
df12

0                         1  \
0     01:35:46             kayalvizhi S:   
1     01:36:00               Rubi Singh:   
2          LMS                      None   
3     01:36:01              Km Muskan .:   
4     01:36:10           PRATIK PATIDAR:   
...        ...                       ...   
1150  01:41:52  Gurram Ruthvi Chaitanya:   
1151  01:41:53          Rahul Janjirala:   
1152  01:41:53                Vyom Modi:   
1153  01:41:53     Mohammed Abdul Rafay:   
1154  01:41:54       Gayathri Kondepudi:   

                                                      2  
0     where can I find the recorded session this mee...  
1                 Replying to "where can I find the..."  
2                                                  None  
3     sir at the end of the class plz once again exp...  
4                 Replying to "where can I find the..."  
...                                                 ...  
1150                      sir it is showing preequation  
1151                             innomatics hd logo sir  
1152  sir should we add experience related to other ...  
1153                                   sir discord link  
1154                   lms access mail not received sir  

[1155 rows x 3 columns]

In [28]:
df13 =  pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (14).txt", sep="\t", header=None, engine='python')

In [29]:
df13

0                      1  \
0     00:16:50         Swastik Dubey:   
1     00:16:50          Manda Divija:   
2     00:16:51         Sai Sudhane G:   
3     00:16:51         Kartik Dhanai:   
4     00:16:51          Sakshi Yadav:   
...        ...                    ...   
2462  00:59:54          Sid Farakate:   
2463  00:59:56   Suresh Bommeraboina:   
2464  00:59:57   RAJ KUMAR SAGGURTHI:   
2465  00:59:59       Nicherla Bharat:   
2466  00:59:59  Patlolla Pranavkumar:   

                                                      2  
0                                                   Yes  
1                                                   yes  
2                                                   Yes  
3                                               Tabaelu  
4                                                   yes  
...                                                 ...  
2462                                       explaination  
2463                             To explore the project  
2464  sir , do we have to share our resume in Linked...  
2465                         Description of the project  
2466                   recuiter can see the our project  

[2467 rows x 3 columns]

In [33]:
df14 =  pd.read_csv("GMT20240125-121850_RecordingnewChat - Copy (15).txt", sep="\t", header=None, engine='python')

In [34]:
df14

0                                    1  \
0     00:17:09                        Rahul Durgam:   
1     00:17:09                         Abdul Fazil:   
2     00:17:09                      Hridya Shree R:   
3     00:17:09                        Ayush Poddar:   
4     00:17:09                          Ravi Singh:   
...        ...                                  ...   
3511  01:41:32  Satya Prakash Veera Venkata Gokeda:   
3512  01:41:32                Mohammed Abdul Rafay:   
3513  01:41:33                           Reshma Sk:   
3514  01:41:33                       Koushik Yadav:   
3515  01:41:35                       Prajwal Damre:   

                                   2  
0               software engineering  
1                                 se  
2                                 SE  
3               Software engineering  
4                                 SE  
...                              ...  
3511                   Thank you sir  
3512  sir link to of discord channel  
3513                   Thank you sir  
3514                             kkk  
3515     do we have to submit task??  

[3516 rows x 3 columns]

# **Checking linked emails in DataFrame 1.**

In [45]:
df_count = df[2].astype(str).str.startswith("https://www.linkedin.com").sum()
print(df_count)


490


In [50]:
ab={1,"https://www.linkedin.com/in/kandukuri-jaswanth-61a8a2209/"}

In [51]:
d=pd.DataFrame(ab)

In [52]:
d

0
0                                                  1
1  https://www.linkedin.com/in/kandukuri-jaswanth...

In [54]:
import pandas as pd
import requests

for i in d[0]:  
    if pd.notna(i) and str(i).startswith("https://www.linkedin.com"):  
        try:
            req = requests.get(i, timeout=5)  # Adding a timeout to avoid long waits
            print(f"URL: {i} | Status Code: {req.status_code}")
        except requests.RequestException as e:
            print(f"Failed to fetch {i}: {e}")  # Handling exceptions

URL: https://www.linkedin.com/in/kandukuri-jaswanth-61a8a2209/ | Status Code: 999


# **Checking links and determining the number of valid links across all DataFrames.**

In [35]:
import requests
import pandas as pd

# List of all 15 DataFrames
dataframes = [df1, df2, df3, df4, df5, df6, df7, df8, df9, df10, df11, df12, df13, df14]

# Initialize counters and error tracking
link_count = 0
valid_link_count = 0
fake_links = 0
unique_errors = set()  # Store unique error messages

# Iterate through all DataFrames
for df44 in dataframes:
    for link in df44[2].fillna('').astype(str):  # Ensure column 2 exists
        if link.startswith("https://www.linkedin.com"):
            link_count += 1  # Count total LinkedIn links
            
            try:
                req = requests.get(link, timeout=5)  # Set timeout
                if req.status_code == 200:
                    valid_link_count += 1  # Valid LinkedIn link
                else:
                    fake_links += 1  # Non-200 status means fake or broken link
            except requests.exceptions.RequestException as e:  
                fake_links += 1  
                unique_errors.add(str(e))  # Track unique errors

# Print results
print(f"Total LinkedIn Links Found: {link_count}")
print(f"Valid LinkedIn Links: {valid_link_count}")
print(f"Fake or Unreachable Links: {fake_links}")
print(f"Unique Issues Detected: {len(unique_errors)}")
print(f"Unique Error Messages: {unique_errors}")  # Optional: Print all unique errors

Total LinkedIn Links Found: 5184
Valid LinkedIn Links: 130
Fake or Unreachable Links: 5054
Unique Issues Detected: 0
Unique Error Messages: set()


# **Conclusion**
# An analysis of 15 DataFrames revealed 5,184 LinkedIn links, out of which 130 were valid, while 5,054 were either fake or unreachable. No unique request errors were recorded, suggesting that all exceptions were handled effectively during the validation process, ensuring robust error management and accurate classification of valid and invalid links